### PR TITLE
`Lectures`: Allow trigger transcription for existing videos

### PR DESCRIPTION
**Pull request by Anishyou**
#### General

- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data (e.g. using paging).
- [x] I **strictly** followed the principle of **data economy** for all client-server REST calls.
- [x] I **strictly** followed the [client coding guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] I **strictly** followed the [AET UI-UX guidelines](https://ls1intum.github.io/ui-ux-guidelines/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Description

Allows user to check whether a Transcript is possible for existing videos , if yes user can start the transcript generation. 

### Steps for Testing

Prerequisites:
- Nebula running 
- Add this in you application-local.yml
nebula:
  enabled: true
  url: http://localhost:3007/
  secret-token: "nebula-secret"
tum-live:
  api-base-url: https://tum.live/api/v2

-run Nebula on edutelligence https://github.com/ls1intum/edutelligence
- 1 Instructor

1. Log in to Artemis
2. create a lecture unit with a public tum live video https://live.rbg.tum.de/w/WiSe24ItP/55920
 (Do not check the "generate transcript after saving" checkbox)
3. go to the lecture unit overview as shown in the picture - Transcription possible badge should show
4. go to edit lecture unit , generate transcript after saving checkbox should show. 

To test-
  - With no transcription started: Badge should appear when playlist URL is available
  - With transcription in progress (jobId exists): Badge should NOT appear
  - With completed transcription: Badge should NOT appear (transcript segments are available)
   
   
### Testserver States
You can manage test servers using [Helios](https://helios.aet.cit.tum.de/). Check environment statuses in the [environment list](https://helios.aet.cit.tum.de/repo/69562331/environment/list). To deploy to a test server, go to the [CI/CD](https://helios.aet.cit.tum.de/repo/69562331/ci-cd) page, find your PR or branch, and trigger the deployment.

#### Code Review
- [x] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [x] Test 1
- [ ] Test 2

### Screenshots

image
image
https://github.com/user-attachments/assets/1b4654ae-556b-4a4f-a637-5c95fb026d78

## Summary by CodeRabbit

* **New Features**
  * Shows a "Transcription possible" badge for video units and auto-enriches playlist URLs when opening video unit forms; transcription generation enabled when a playlist URL is present.

* **Behavior**
  * Playlist URL enrichment now happens asynchronously and transcription fetching is gated to admin users.

* **Tests**
  * Expanded coverage for playlist URL enrichment, transcription enablement, badge display, admin gating, and error/edge cases.

* **Documentation**
  * Added English and German labels; removed guidance about pasting transcription JSON as a fallback.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: src/main/webapp/app/lecture/overview/course-lectures/attachment-video-unit/attachment-video-unit.component.html [hunk 1]
```diff
@@ -43,6 +43,14 @@
     }
 
     @if (!!videoUrl()) {
+        @if (playlistUrl() && canShowTranscriptionPossibleBadge()) {
+            <div class="mb-3">
+                <span class="badge bg-info">
+                    <fa-icon [icon]="faFileLines" class="me-1" />
+                    <span jhiTranslate="artemisApp.attachmentVideoUnit.transcriptAvailable"></span>
```
[the remark this answers is not in the document]
Comment (Anishyou) [reply src/main/webapp/app/lecture/overview/course-lectures/attachment-video-unit/attachment-video-unit.component.html:67]: it is actually correct

File: src/main/webapp/app/lecture/overview/course-lectures/attachment-video-unit/attachment-video-unit.component.ts [hunk 2]
```diff
@@ -60,10 +62,34 @@ export class AttachmentVideoUnitComponent extends LectureUnitDirective<Attachmen
...
+    readonly canShowTranscriptionPossibleBadge = computed(() => {
+        // Don't show badge if transcription was created (which would have a jobId)
+        if (this.lectureUnit().transcriptionProperties) {
+            return false;
+        }
+        // Don't show badge if transcript is already available
+        if (this.hasTranscript()) {
+            return false;
+        }
+        return true;
```
Comment (florian-glombik) [inline src/main/webapp/app/lecture/overview/course-lectures/attachment-video-unit/attachment-video-unit.component.ts]: I think this could be a variable 


```suggestion
        // noinspection UnnecessaryLocalVariableJS: not inlined because the variable name improves readability
        const isTranscriptAlreadyAvailable = this.hasTranscript();
        return isTranscriptAlreadyAvailable;
```

File: src/main/webapp/app/lecture/manage/lecture-units/attachment-video-unit-form/attachment-video-unit-form.component.ts [hunk 3]
```diff
@@ -211,14 +221,29 @@ export class AttachmentVideoUnitFormComponent implements OnChanges {
...
             this.transcriptionStatus.set(newStatus);
+           // Set playlist URL if available from formData (for existing videos)
+            if (formData.playlistUrl) {
+                this.playlistUrl.set(formData.playlistUrl);
+                this.canGenerateTranscript.set(true);
+            }
         } else {
             this.transcriptionStatus.set(undefined);
+
         }
```
Comment (florian-glombik) [inline src/main/webapp/app/lecture/manage/lecture-units/attachment-video-unit-form/attachment-video-unit-form.component.ts]: Wouldn't it make sense to directly only define this in the effect? Right now it seems to me as if we have duplicated logic there. 

With the current approach we are triggering the effect and the ngOnChanges if formData() changes, don't we?

File: src/main/webapp/app/lecture/manage/lecture-units/lecture-units.component.ts [hunk 4]
```diff
@@ -361,6 +361,23 @@ export class LectureUpdateUnitsComponent implements OnInit {
...
+                                            ...this.attachmentVideoUnitFormData,
+                                            playlistUrl: playlist,
+                                        };
+                                    }
+                                },
+                                error: () => {
+                                    // Playlist URL not available, transcription generation won't be possible
+                                },
+                            });
+                        }
```
Comment (florian-glombik) [inline src/main/webapp/app/lecture/manage/lecture-units/lecture-units.component.ts]: This appears to be duplicated code with `src/main/webapp/app/lecture/manage/lecture-units/edit-attachment-video-unit/edit-attachment-video-unit.component.ts` - can we extract it?

Maybe something like this in `attachment-video-unit.service.ts` would make sense?
```
    /**
     * Fetches playlist URL for a video source and updates the form data if found.
     * This is a helper method to reduce code duplication when setting up video units for editing.
     *
     * @param videoSource - The video source URL to fetch playlist for
     * @param currentFormData - The current form data to update
     * @returns Observable that emits the updated form data with playlist URL, or original form data if not found
     */
    fetchAndUpdatePlaylistUrl<T extends { playlistUrl?: string }>(videoSource: string | undefined, currentFormData: T): Observable<T> {
        if (!videoSource) {
            return of(currentFormData);
        }

        return this.getPlaylistUrl(videoSource).pipe(
            map((playlist) => {
                if (playlist) {
                    return { ...currentFormData, playlistUrl: playlist };
                }
                return currentFormData;
            }),
        );
    }
```

Which would then also simplify the code here?
```
                  this.attachmentVideoUnitService.fetchAndUpdatePlaylistUrl(this.attachmentVideoUnit.videoSource, this.formData).subscribe({
                        next: (updatedFormData) => {
                            this.formData = updatedFormData;
                        },
                    });
```

File: src/main/webapp/app/lecture/overview/course-lectures/attachment-video-unit/attachment-video-unit.component.spec.ts [hunk 5]
```diff
@@ -386,4 +386,44 @@ describe('AttachmentVideoUnitComponent', () => {
...
+        component.lectureUnit().transcriptionProperties = {
+            lectureUnitId: 1,
+            language: 'en',
+            segments: [],
+        };
+        component.transcriptSegments.set([{ startTime: 0, endTime: 1, text: 'test', slideNumber: 1 }]);
+        fixture.detectChanges();
+
+        expect(component.canShowTranscriptionPossibleBadge()).toBeFalse();
+    });
```
Comment (florian-glombik) [inline src/main/webapp/app/lecture/overview/course-lectures/attachment-video-unit/attachment-video-unit.component.spec.ts]: Can we group the tests in a describe?

**Review by florian-glombik — CHANGES_REQUESTED** [review]
I think the current implementation in src/main/webapp/app/lecture/manage/lecture-units/attachment-video-unit-form/attachment-video-unit-form.component.ts is not optimal with the effect and onChanges. I think we should only use the effect there and updated it accordingly.

I also think that have some code duplicates that we can get rid off 😄

**Review by florian-glombik — COMMENTED** [review]
## Change Request within this PR
I cannot save a transcript anymore, I can only save if I choose to generate the transcript

Image

https://github.com/user-attachments/assets/74364f99-c6a4-4659-97ef-e9e0e0a3f25c

## Future Improvements
For future PRs: I think that the UI is rather unintuitive, can we improve that? 
* I have to open the lecture unit to see that the transcript generation failed, I think I should see this information outside of that view, I should not need to open the lecture unit Image
* Same goes for the unit view, from the first picture I would not assume that something went wrong
     Image
   Image

**Review by florian-glombik — CHANGES_REQUESTED** [review]
https://github.com/ls1intum/Artemis/pull/11675#pullrequestreview-3493895899

**Review by florian-glombik — APPROVED** [review]
Works as described, thanks for the code changes and help to setup nebula 🙌 

While testing it I noticed that the Lecture Units Feature is in a state that leaves plenty of room for improvement from a UI/UX perspective - I created respective issues:
* https://github.com/ls1intum/Artemis/issues/11679
* https://github.com/ls1intum/Artemis/issues/11680
